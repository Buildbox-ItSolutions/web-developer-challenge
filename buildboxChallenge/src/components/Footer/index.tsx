import React from 'react'
import { Button, Image } from '..'
import { FooterContainer, FooterCredits } from './style'

export function Footer() {
  return (
    <>
    <FooterContainer>
            <div className='footer-container__wrap'>
                    <div className='footer-content__logo'>
                          <Image src={'https://buildbox.com.br/site-build/wp-content/themes/buildbox/assets/img/buildbox-black.svg'} alt='logo' behavior="contain" />
                    </div>
                    <div className='footer-content__navigation'>
                        <ul className='footer-menu'>
                            <li>In√≠cio</li>
                            <li>Ideias</li>
                            <li>O que fazemos</li>
                            <li>Contato</li>
                            <li>Trabalhos</li>
                            <li>Carreiras</li>
                        </ul>
                        <ul className='footer-menu'>
                            <li>Facebook</li>
                            <li>O que fazemos</li>
                            <li>Ideias</li>
                            <li>Contato</li>
                            <li>Carreiras</li>
                            <li>Trabalhos</li>
                        </ul>
                        <ul className='footer-menu footer-menu-3'>
                            <li>Instagram</li>
                            <li>Linkedin</li>
                            <li>Investidores</li>
                            <li>Ideias</li>
                            <li>Events</li>
                            <li>Facebook</li>
                        </ul>
                    </div>
                    <div className='footer-content__newsletter'>
                      <div>
                        <h3 className='subscribe'>Receba nossa newsletter</h3>
                        <div className='input-text'>
                            <input type="text" placeholder='Seu email'/>
                        </div>
                        <div className='btn-subscribe'>
                          <Button >Subscribe</Button>
                        </div>
                      </div>
                    </div>
            </div>
        </FooterContainer>
        <FooterCredits>
            <div className='footer-credits__wrap'>
                <div className='footer-credits__wrap'>
                    <div className='footer-credits__info'>
                        <p>Developer Wesley Ferreira</p>
                        <p>Terms and Conditions | Privacy Policy</p>
                    </div>
                
                </div>
            </div>
        </FooterCredits>
    </>
    )
}
