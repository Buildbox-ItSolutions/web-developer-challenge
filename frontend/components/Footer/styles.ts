import styled from 'styled-components';

export const Footer = styled.footer`
    width: 100%;
    height: 100px;
    grid-area: footer;
    background-color: #2b2b2b;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    @media (max-width: 342px) {
        text-align: center;
    }
`;

export const P = styled.p`
    color: #9f9f9f;
    font-size: 1em;
`;
