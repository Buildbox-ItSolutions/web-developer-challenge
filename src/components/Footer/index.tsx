import styled  from "styled-components";

const Content = styled.div`
    
`;

export default function Footer() {
    return (
        <Content>
            
        </Content>
    );
}