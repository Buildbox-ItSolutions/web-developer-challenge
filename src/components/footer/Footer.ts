import styled from "styled-components";

export const FooterContainer = styled.div`
  display: flex;
  justify-content: center;
  padding: 10px;
  gap: 1rem;
  width: 100%;
`;

export const Link = styled.a`
  cursor: pointer;
`;