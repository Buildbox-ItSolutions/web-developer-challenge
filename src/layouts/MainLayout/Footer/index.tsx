import { Typography } from '@mui/material'
import { Stack } from '@mui/system'
import React from 'react'
import BxLogo from '../../../assets/BxLogo'
import Link from '../../../components/Link'

export default function Footer() {
  return (
    <Stack
      bottom={0}
      alignItems={'flex-start'}
      paddingLeft={8}
      px={{xs: 2, sm: 4}}
      py={8}
      justifyContent={'flex-start'}
      height={'100%'}
      bgcolor={'#2b2b2b'}
    >

      <Stack width={'100%'} justifyContent={'space-between'} flexDirection={'row'} >
        <Stack flex={1}>
          <BxLogo />
          <Typography color={'#9f9f9f'} variant={'caption'} >Challenge Post buildBox by:</Typography>
          <Typography color={'#9f9f9f'} variant={'button'} > Mateus Martins</Typography>
        </Stack>
        <Stack flex={1} alignItems={{xs: 'flex-end', sm: 'flex-start'}} >
          <Typography color={'#9f9f9f'} variant={'button'} >Cr√©ditos</Typography>
          <Link href="https://www.flaticon.com/free-icons/box" title="box icons">Box icons created by Freepik - Flaticon</Link>
        </Stack>
      </Stack>
    </Stack>
  )
}
