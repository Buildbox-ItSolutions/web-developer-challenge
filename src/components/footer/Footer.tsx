import { FooterContainer, Link } from './Footer'
import linkedinLogo from '../../assets/linkedin.svg'
import gitHubLogo from '../../assets/github.svg'


function Footer() {

  return (
    <FooterContainer>
      <Link href="https://github.com/cleitson" target="_blank" rel="noreferrer">
        <img src={gitHubLogo} alt="perfil github" />
      </Link>
      <Link href="https://www.linkedin.com/in/cleitsonlima/" target="_blank" rel="noreferrer">
        <img src={linkedinLogo} alt="perfil linkedin" />
      </Link>
    </FooterContainer>
  )
}

export default Footer