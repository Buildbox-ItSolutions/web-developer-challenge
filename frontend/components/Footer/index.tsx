import { FC } from 'react';

import { Footer, P } from './styles';

const FooterComponent: FC = () => {
    return (
        <Footer>
            <P>&copy; Copyright 2019 - 2020.</P>
            <P>
                All rights reserved. Powered by the <strong>Buildbox</strong>
            </P>
        </Footer>
    );
};

export default FooterComponent;
