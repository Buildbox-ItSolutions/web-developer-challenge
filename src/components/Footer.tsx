import styles from '../styles/Footer.module.css';

export default function Footer() {
  return (
    <div className={ styles.container }>Create By: Yuri Salgado Farias Torres</div>
  );
}
