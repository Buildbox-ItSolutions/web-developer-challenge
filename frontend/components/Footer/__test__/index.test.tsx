import 'jest-styled-components';

import React from 'react';
import renderer from 'react-test-renderer';

import Footer from '../index';

describe('Footer', () => {
    it('must render without errors', () => {
        const wrapper = renderer.create(<Footer />).toJSON();

        expect(wrapper).toMatchSnapshot();
    });
});
