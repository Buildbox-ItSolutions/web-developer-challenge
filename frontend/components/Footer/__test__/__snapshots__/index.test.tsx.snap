// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer must render without errors 1`] = `
.c0 {
  width: 100%;
  height: 100px;
  grid-area: footer;
  background-color: #2b2b2b;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  color: #9f9f9f;
  font-size: 1em;
}

@media (max-width:342px) {
  .c0 {
    text-align: center;
  }
}

<footer
  className="c0"
>
  <p
    className="c1"
  >
    Â© Copyright 2019 - 2020.
  </p>
  <p
    className="c1"
  >
    All rights reserved. Powered by the 
    <strong>
      Buildbox
    </strong>
  </p>
</footer>
`;
